File: frontend/src/constants/categories.ts
/**
 * Authoritative issue category definitions for CommitRush.
 * Mirrors backend/core/categories.py.
 */

export interface IssueCategory {
  value: string;
  label: string;
  count?: number;
}

export const ISSUE_CATEGORIES: IssueCategory[] = [
  { value: 'feature', label: 'Features' },
  { value: 'test', label: 'Testing & QA' },
  { value: 'bug', label: 'Bug Fixes' },
  { value: 'security', label: 'Security' },
  { value: 'performance', label: 'Performance' },
  { value: 'docs', label: 'Documentation' },
  { value: 'refactor', label: 'Refactoring' },
  { value: 'validation', label: 'Validation' },
  { value: 'ui', label: 'UI' },
  { value: 'database', label: 'Database & Storage' },
  { value: 'networking', label: 'Networking' },
  { value: 'frontend', label: 'Frontend' },
  { value: 'backend', label: 'Backend' },
  { value: 'devops', label: 'DevOps / Infrastructure' },
  { value: 'fullstack', label: 'Fullstack' },
  { value: 'dx', label: 'Developer Experience' },
];

export const CATEGORY_LABEL_MAP: Record<string, string> = Object.fromEntries(
  ISSUE_CATEGORIES.map((c) => [c.value, c.label])
);

File: backend/core/categories.py
"""
CommitRush Authoritative Issue Category Definitions.
Single source of truth for issue categories, user-facing labels, and serializer choices.
"""

ISSUE_CATEGORIES: list[tuple[str, str]] = [
    ('feature', 'Features'),
    ('test', 'Testing & QA'),
    ('bug', 'Bug Fixes'),
    ('security', 'Security'),
    ('performance', 'Performance'),
    ('docs', 'Documentation'),
    ('refactor', 'Refactoring'),
    ('validation', 'Validation'),
    ('ui', 'UI'),
    ('database', 'Database & Storage'),
    ('networking', 'Networking'),
    ('frontend', 'Frontend'),
    ('backend', 'Backend'),
    ('devops', 'DevOps / Infrastructure'),
    ('fullstack', 'Fullstack'),
    ('dx', 'Developer Experience'),
]

CATEGORY_CHOICES = ISSUE_CATEGORIES
CATEGORY_MAP = dict(ISSUE_CATEGORIES)
VALID_CATEGORY_KEYS = [k for k, _ in ISSUE_CATEGORIES]

DEFAULT_CATEGORY_MULTIPLIERS: dict[str, float] = {
    'feature': 1.5,
    'bug': 1.0,
    'docs': 0.8,
    'test': 1.2,
    'security': 1.5,
    'performance': 1.3,
    'refactor': 1.0,
    'validation': 1.0,
    'ui': 1.1,
    'database': 1.2,
    'networking': 1.2,
    'frontend': 1.1,
    'backend': 1.2,
    'devops': 1.2,
    'fullstack': 1.3,
    'dx': 1.0,
}

